pkg/builder/mutating_webhook_test.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package builder_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	pkgbuilder "github.com/vmware-tanzu/vm-operator/pkg/builder"
+	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+	pkgctxfake "github.com/vmware-tanzu/vm-operator/pkg/context/fake"
+)
+
+var _ = Describe("NewMutatingWebhook", func() {
+
+	var (
+		err        error
+		wh         *pkgbuilder.MutatingWebhook
+		whName     string
+		whCallback pkgbuilder.Mutator
+		ctx        *pkgctx.ControllerManagerContext
+		obj        metaRuntimeObj
+		gvk        schema.GroupVersionKind
+		res        admission.Response
+		recorder   record.EventRecorder
+		scheme     *runtime.Scheme
+	)
+
+	BeforeEach(func() {
+		ctx = pkgctxfake.NewControllerManagerContext()
+		obj = getConfigMap()
+		scheme = runtime.NewScheme()
+		whName = "fake"
+		whCallback = &fakeMutator{}
+	})
+
+	AfterEach(func() {
+		err = nil
+		wh = nil
+		ctx = nil
+		gvk = getConfigMapSchemaGVK()
+		res = admission.Response{}
+		recorder = nil
+		scheme = nil
```

Comment:
Nit: Some of these variable resets (e.g. ctx, scheme) are unnecessary here as they will always be initialized in the `BeforeEach`.
Also, could you help explain the purpose of `gvk = getConfigMapSchemaGVK()` here? I am a bit confused about why it uses the ConfigMap schema and why it's in the `AfterEach`...

Reply:
Sure, some are not necessary, but I will likely not resend this to address that since it will require another run. Maybe in a follow-up. The `gvk` is used as part of building the `fakeMutator` and `fakeValidator` -- `gvk` is part of the response that can be set up in advance for testing.

Reply:
I went ahead and addressed these, thank you!